tests/hpc/slurm_master.pm

```diff
@@ -19,38 +19,264 @@ use warnings;
+    script_run("sacctmgr -i add account UNI_Y_Biology Description=\"UNI_Y\" Organization=UNI_Y");
+    #Add users associated with account in org=UNI_Y
+    script_run("sacctmgr -i create user name=Joe DefaultAccount=UNI_Y_Physics");
+    script_run("sacctmgr -i create user name=Noah DefaultAccount=UNI_Y_Biology");
+    script_run("sacctmgr -i create user name=$users{user_4} DefaultAccount=UNI_Y_Physics");
+    script_run("sacctmgr -i create user name=$users{user_3} DefaultAccount=UNI_Y_Biology");
+
+    script_run('sacctmgr show account');
+    script_run('sacctmgr show associations');
+    record_info('INFO', script_run('sacctmgr show account'));
+
+    script_run("srun --uid=$users{user_1} --account=UNI_X_Math -w slave-node00,slave-node01 date");
+    script_run("srun --uid=$users{user_2} --account=UNI_X_IT -N 2 hostname");
+
+    script_run("srun --uid=$users{user_3} --account=UNI_Y_Biology -N 3 date");
+    script_run("srun --uid=$users{user_4} --account=UNI_Y_Physics -N 3 hostname");
+
+    #Yet another sleep. Slurm.conf::JobAcctGatherFrequency=12
+    #In order to allow information to be dumped to the DB, we need to wait some time
+    sleep(30);
+
+    my $jobs = script_output("sacct -n -p --starttime 2010-01-01 --format=User,Account,JobID,Jobname,partition,state,time,start,end,elapsed,MaxRss,MaxVMSize,nnodes,ncpus,nodelist");
+
+    #check if there are expected srun jobs being recorded in the accounting db
+    $result = 1 unless (($jobs =~ /$users{user_1}/) &&
+        ($jobs =~ /$users{user_2}/) &&
+        ($jobs =~ /$users{user_3}/) &&
+        ($jobs =~ /$users{user_4}/));
+
+    ##check the content directly in the db; see: bugzilla#1150565
+    # This is only for information now
+    # TODO: consider adding sanity checks - direct checks in the db - for some tests
+    my $db_elements = script_output("mysql -h slave-node02.openqa.test -uslurm -e \"use slurm_acct_db; select * from linux_job_table;\"");
+    record_info('INFO DB', "$db_elements");
+
+  FAIL:
+    my %results = generate_results($name, $description, $result);
+    return %results;
+}
 
 sub run {
-    my $self  = shift;
-    my $nodes = get_required_var("CLUSTER_NODES");
+    my $self       = shift;
+    my $nodes      = get_required_var('CLUSTER_NODES');
+    my $slurm_conf = get_required_var('SLURM_CONF');
     $self->prepare_user_and_group();
 
-    # install slurm
-    zypper_call('in slurm slurm-munge');
+    # provision HPC cluster, so the proper rpms are installed,
+    # munge key is distributed to all nodes, so is slurm.conf
+    # and proper services are enabled and started
+    zypper_call('in slurm slurm-munge slurm-torque');
 
-    $self->prepare_slurm_conf();
-    barrier_wait("SLURM_SETUP_DONE");
+    #types of slurm set-ups: basic, accounting, ha, nfs_db
+    if ($slurm_conf =~ /ha/) {
+        zypper_call('in nfs-client rpcbind');
+        $self->mount_nfs();
```

Comment:
isn´t this call already installing nfs-client and rpcbind? why is it needed here?